pkg/models/metrics/metrics.go

```diff
@@ -19,785 +19,16 @@
-	for e := range ch {
-		apiResponse = append(apiResponse, e)
-	}
-
-	return &Response{
-		MetricsLevel: MonitorLevelWorkload,
-		Results:      apiResponse,
-	}
-}
-
-func GetPodMetrics(params RequestParams) *Response {
-	client, err := cs.ClientSets().Prometheus()
-	if err != nil {
-		klog.Error(err)
-		return nil
-	}
-
-	ch := make(chan APIResponse, ChannelMaxCapacity)
-	var wg sync.WaitGroup
-
-	// for each metric, make PromQL expression and send the request to Prometheus servers
-	for _, metricName := range podMetrics {
-		matched, _ := regexp.MatchString(params.MetricsFilter, metricName)
-		if matched {
-			wg.Add(1)
-			go func(metricName string, params RequestParams) {
-				exp := makePromqlForPod(metricName, params)
-				v := url.Values{}
-				for key, value := range params.QueryParams {
-					v[key] = value
-				}
-				v.Set("query", exp)
-				response := client.QueryToK8SPrometheus(params.QueryType, v.Encode())
-
-				// add label resouce_name
-				for _, item := range response.Data.Result {
-					item.Metric["resource_name"] = item.Metric["pod_name"]
-				}
-
-				ch <- APIResponse{
-					MetricName:  metricName,
-					APIResponse: response,
-				}
-				wg.Done()
-			}(metricName, params)
-		}
-	}
-	wg.Wait()
-	close(ch)
-
-	var apiResponse []APIResponse
-	for e := range ch {
-		apiResponse = append(apiResponse, e)
-	}
-
-	return &Response{
-		MetricsLevel: MonitorLevelPod,
-		Results:      apiResponse,
-	}
-}
-
-func GetContainerMetrics(params RequestParams) *Response {
-	client, err := cs.ClientSets().Prometheus()
-	if err != nil {
-		klog.Error(err)
-		return nil
-	}
-
-	ch := make(chan APIResponse, ChannelMaxCapacity)
-	var wg sync.WaitGroup
-
-	// for each metric, make PromQL expression and send the request to Prometheus servers
-	for _, metricName := range containerMetrics {
-		matched, _ := regexp.MatchString(params.MetricsFilter, metricName)
-		if matched {
-			wg.Add(1)
-			go func(metricName string, params RequestParams) {
-				exp := makePromqlForContainer(metricName, params)
-				v := url.Values{}
-				for key, value := range params.QueryParams {
-					v[key] = value
-				}
-				v.Set("query", exp)
-				response := client.QueryToK8SPrometheus(params.QueryType, v.Encode())
-
-				// add label resouce_name
-				for _, item := range response.Data.Result {
-					item.Metric["resource_name"] = item.Metric["container_name"]
-				}
-
-				ch <- APIResponse{
-					MetricName:  metricName,
-					APIResponse: response,
-				}
-				wg.Done()
-			}(metricName, params)
-		}
-	}
-	wg.Wait()
-	close(ch)
-
-	var apiResponse []APIResponse
-	for e := range ch {
-		apiResponse = append(apiResponse, e)
-	}
-
-	return &Response{
-		MetricsLevel: MonitorLevelContainer,
-		Results:      apiResponse,
-	}
-}
-
-func GetPVCMetrics(params RequestParams) *Response {
-	client, err := cs.ClientSets().Prometheus()
+func GetMetrics(opt monitoring.QueryOptions) (*monitoring.MetricsGroup, error) {
+	client, err := cs.ClientSets().MonitoringClient()
```

Comment:
dependency should be passed from caller, we shouldn't use client factory any more